assets/scripts/app.js

```diff
@@ -0,0 +1,245 @@
+    }
+    return parsedValue;
+}
+
+try {
+    choosenMaxLife = getMaxLifeValue();
+} catch (error) {
+    console.log(error);
+    choosenMaxLife = 100;
+    alert('Invalid input, default value 100 was selected ');
+}
+
+let currentMonsterHealth = choosenMaxLife;
+let currentPlayerHealth = choosenMaxLife;
+let hasBonus = true;
+
+
+adjustHealthBars(choosenMaxLife);
+
+
+function reset(){
+    currentMonsterHealth = choosenMaxLife;
+    currentPlayerHealth = choosenMaxLife;
+    resetGame(choosenMaxLife);
+}
+
+
+function endRound(){
+    const initialPlayerHealth = currentPlayerHealth
+    const playerDamage = dealPlayerDamage(MONSTER_ATTACK_VALUE);
+    currentPlayerHealth -= playerDamage;
+
+    writeToLog(
+        LOG_EVENT_MONSTER_ATTACK,
+        playerDamage,
+        currentMonsterHealth,
+        currentPlayerHealth
+    );
+
+    if (currentPlayerHealth <= 0 && hasBonus){
+        hasBonus = false;
+        removeBonusLife();
+        currentPlayerHealth = initialPlayerHealth;
+        setPlayerHealth(initialPlayerHealth);
+        alert ('You would have died but bonus life saved you!')
+    }
+
+    if (currentMonsterHealth <= 0 && currentPlayerHealth > 0){
+        alert('You Won!');
+
+        writeToLog(
+            LOG_EVENT_GAME_OVER,
+            'YOU WON',
+            currentPlayerHealth,
+            currentMonsterHealth
+        )
+
+    } else if (currentPlayerHealth <=0 && currentMonsterHealth > 0){
+        alert('You Lost!');
+
+        writeToLog(
+            LOG_EVENT_GAME_OVER,
+            'MONSTER WON',
+            currentPlayerHealth,
+            currentMonsterHealth
+        )
+
+    } else if ( currentMonsterHealth <= 0 && currentPlayerHealth <=0){
+        alert('You have a draw!')
+
+        writeToLog(
+            LOG_EVENT_GAME_OVER,
+            'A DRAW',
+            currentPlayerHealth,
+            currentMonsterHealth
+        )
+    }
+
+    if (currentPlayerHealth <= 0 || currentMonsterHealth <= 0){
+        reset();
+    }
+}
+
+
+function attackMonster(mode){
+    let maxDamage;
+    let logEvent
+    if (mode === MODE_ATTACK){
+        maxDamage = ATTACK_VALUE;
+        logEvent = LOG_EVENT_PLAYER_ATTACK;
+    } else if (mode === MODE_STRONG_ATTACK){
+        maxDamage = STRONG_ATTACK_VALUE;
+        logEvent = LOG_EVENT_PLAYER_STRONG_ATTACK;
+    }
+    const damage = dealMonsterDamage(maxDamage);
+    currentMonsterHealth -= damage;
+
+    writeToLog(
+        logEvent,
+        damage,
+        currentPlayerHealth,
+        currentMonsterHealth
+    );
+
+    endRound();
+
+}
+function attackHandler(){
+    attackMonster(MODE_ATTACK)
+}
+function strongAttackHandler(){
+    attackMonster(MODE_STRONG_ATTACK)
+}
+
+
+function healHandler(){
+    let healValue;
+    if (currentPlayerHealth >= choosenMaxLife - HEAL_VALUE){
+        alert("You can't heal more.")
+        healValue = choosenMaxLife - currentPlayerHealth;
+    } else{
+        healValue = HEAL_VALUE;
+    }
```

Comment:
can you use ternary operator here ? @mmusk

Reply:
i would prefer if..else because here its more than 2 statements

Reply:
yeah sure

Reply:
Thanks @faran4engg